### DISCUSSION TITLE
Load Client Assemblies

### DISCUSSION THREAD

**Discussion by lthoa** [oqtane/oqtane.framework#996] [Q&A]
How can I load third-party component dll files in a Module project?
In my case, I use [MediatR](https://github.com/jbogard/MediatR) on the client and server side.

Thanks.

**Comment by sbwalker** [comment]
Your module will need to implement the IModule interface and specify the Dependencies property ( it is a comma delimited string which must contain all of the dependent assemblies which your module relies upon ). Once you include this specification, Oqtane will automatically download the assemblies to the client browser for you.

https://github.com/oqtane/oqtane.blogs/blob/master/Client/ModuleInfo.cs

**Reply by lthoa** [reply]
Great, 
Thank you very much @sbwalker!

**Reply by sbwalker** [reply]
I assume you also already set the setting in Oqtane.Server\appsettings.json:

```
{
  "ConnectionStrings": {},
  "Runtime": "WebAssembly",
  ...
}
```

**Reply by lthoa** [reply]
And we need create **ClientStartup.cs**

public class ClientStartup : IClientStartup{
        public void ConfigureServices(IServiceCollection services)
        {                        
            services.AddService...(...);
        }
}
and **Startup.cs** on server side
public class Startup : StartupBase{
        public override void ConfigureServices(IServiceCollection services)
        {
        .....
        }
}

**Reply by sbwalker** [reply]
You do not need to implement IClientStartup or IServerStartup if you decorate your services with the IService interface. IService allows the framework to do auto service registration. It is the approach used in the Blog module example, the Module Creator, and the module development video tutorial.

**Reply by markdav-is** [reply]
you can wrap your code in three ticks to have it formatted like code if you like 
``` example code wrapped with three 's ```
